eng/testing/tests.targets

```diff
@@ -115,6 +115,69 @@
+    </PropertyGroup>
+    <!-- We need to copy additional stuff into $(OutDir)\Bundle
+         1) The whole BCL
+         2) Test Runner (with xharness client-side lib)
+      -->
+    <ItemGroup>
+      <TestBinaries Include="$(OutDir)\*.*"/>
+      <AppleTestRunnerBinaries Include="$(AppleTestRunner)\*.*" />
+      <BclBinaries Include="$(RuntimePackDir)\lib\$(NetCoreAppCurrent)\*.*" Exclude="$(RuntimePackDir)\lib\$(NetCoreAppCurrent)\System.Runtime.WindowsRuntime.dll" />
+      <BclBinaries Include="$(RuntimePackDir)\native\*.*" />
+    </ItemGroup>
+    <Error Condition="!Exists('$(AppleTestRunner)')" Text="AppleTestRunner=$(AppleTestRunner) doesn't exist" />
+    <Error Condition="!Exists('$(RuntimePackDir)')" Text="RuntimePackDir=$(RuntimePackDir) doesn't exist" />
+    <RemoveDir Directories="$(BundleDir)" />
+    <Copy SourceFiles="@(TestBinaries)" DestinationFolder="$(BundleDir)" SkipUnchangedFiles="true"/>
+    <Copy SourceFiles="@(AppleTestRunnerBinaries)" DestinationFolder="$(BundleDir)\%(RecursiveDir)" SkipUnchangedFiles="true"/>
+    <Copy SourceFiles="@(BclBinaries)" DestinationFolder="$(BundleDir)\%(RecursiveDir)" SkipUnchangedFiles="true"/>
+
+    <!-- xharness expects 'xunit-excludes.txt' with xunit categories to ignore in the bundle -->
+    <ItemGroup>
+      <XunitIgnoreCategories Include="category=OuterLoop"/>
+      <XunitIgnoreCategories Include="category=failing"/>
+      <XunitIgnoreCategories Include="category=nonnetcoreapptests"/>
+      <XunitIgnoreCategories Include="category=nonosxtests"/>
+      <XunitIgnoreCategories Include="category=nonmonotests"/>
+    </ItemGroup>
+    <WriteLinesToFile
+        File="$(BundleDir)\xunit-excludes.txt"
+        Lines="@(XunitIgnoreCategories)"
+        Overwrite="true"
+        Encoding="Unicode"/>
+
+    <!-- Run App bundler, it should AOT libs (if needed), link all native bits, compile simple UI (written in ObjC)
+         and produce an app bundle (with xcode project) -->
+    <AppleAppBuilderTask 
+        Arch="$(TargetArchitecture)"
+        ProjectName="$(AssemblyName)"
+        MonoRuntimeHeaders="$(RuntimePackDir)\native\include\mono-2.0"
+        CrossCompiler="$(RuntimePackDir)\native\cross\mono-aot-cross"
+        MainLibraryFileName="AppleTestRunner.dll"
+        UseConsoleUITemplate="True"
+        GenerateXcodeProject="True"
+        BuildAppBundle="True"
+        DevTeamProvisioning="$(DevTeamProvisioning)"
+        OutputDirectory="$(BundleDir)"
+        AppDir="$(BundleDir)">
+        <Output TaskParameter="AppBundlePath" PropertyName="AppBundlePath" />
+        <Output TaskParameter="XcodeProjectPath" PropertyName="XcodeProjectPath" />
+    </AppleAppBuilderTask>
+    <Message Importance="High" Text="Xcode: $(XcodeProjectPath)"/>
+    <Message Importance="High" Text="App: $(AppBundlePath)"/>
+    <!-- This app is now can be consumed by xharness CLI to deploy on a device or simulator -->
+  </Target>
+  <UsingTask TaskName="AppleAppBuilderTask" 
```

Comment:
Please move that one above the target in which it is used. That's our usual pattern.